packages/babel-parser/src/parse-error.ts

```diff
@@ -57,44 +74,39 @@ function toParseErrorConstructor<ErrorDetails extends object>({
   };
 
   return function constructor({ loc, details }: ConstructorArgument) {
-    return instantiate(
-      SyntaxError,
-      { ...properties, loc },
-      {
-        clone(
-          overrides: {
-            loc?: Position;
-            details?: ErrorDetails;
-          } = {},
-        ) {
-          const loc = (overrides.loc || {}) as Partial<Position>;
-          return constructor({
-            loc: new Position(
-              "line" in loc ? loc.line : this.loc.line,
-              "column" in loc ? loc.column : this.loc.column,
-              "index" in loc ? loc.index : this.loc.index,
-            ),
-            details: { ...this.details, ...overrides.details },
-          });
-        },
-        details: { value: details, enumerable: false },
-        message: {
-          get(this: ConstructorArgument): string {
-            return `${toMessage(this.details)} (${this.loc.line}:${
-              this.loc.column
-            })`;
-          },
-          set(value: string) {
-            Object.defineProperty(this, "message", { value });
-          },
-        },
-        pos: { reflect: "loc.index", enumerable: true },
-        missingPlugin: "missingPlugin" in details && {
-          reflect: "details.missingPlugin",
-          enumerable: true,
-        },
+    const error = new SyntaxError();
+    Object.assign(error, properties, { loc, pos: loc.index });
+    if ("missingPlugin" in details) {
+      Object.assign(error, { missingPlugin: details.missingPlugin });
+    }
+
+    type Overrides = {
+      loc?: Position;
+      details?: ErrorDetails;
+    };
+    defineHidden(error, "clone", function clone(overrides: Overrides = {}) {
+      const { line, column, index } = overrides.loc ?? loc;
+      return constructor({
+        loc: new Position(line, column, index),
+        details: { ...details, ...overrides.details },
+      });
+    });
+
+    defineHidden(error, "details", details);
+
+    Object.defineProperty(error, "message", {
+      configurable: true,
+      get(this: ParseError<ErrorDetails>): string {
+        const message = `${toMessage(details)} (${loc.line}:${loc.column})`;
+        this.message = message;
+        return message;
+      },
+      set(value: string) {
+        Object.defineProperty(this, "message", { value, writable: true });
```

Comment:
This was missing `writable: true`, causing the error in https://github.com/babel/babel/pull/9509.